client/lib/utils/time-ago.js

```diff
@@ -0,0 +1,48 @@
+import { translate as __ } from 'lib/mixins/i18n';
+import { sprintf } from 'sprintf-js';
+
```

Comment:
I suggest you take a look at `date-fns`: https://date-fns.org/docs/distanceInWordsToNow
I18N in `date-fns`: https://date-fns.org/docs/I18n

It's being advertised as "similar to `moment`, but you can just use individual functions". It certainly looks like a better solution than reimplementing it yourself :)